libs/IO/Image.h

```diff
@@ -107,6 +109,49 @@ class IO_API CImage
 	#endif
 
 protected:
+    template<typename T>
+    static void findMinMaxPercentile(void *data, Size size, T* min, T* max){
+        if (size == 0){
+            *min = *max = 0;
+            return;
+        }
+
+        // Find min/max
+        T *p = reinterpret_cast<T *>(data);
+        T aMin = p[0];
+        T aMax = p[0];
+
+        for (Size i = 1; i < size; i++){
+            if (p[i] > aMax) aMax = p[i];
+            if (p[i] < aMin) aMin = p[i];
+        }
+
+        double range = static_cast<double>(aMax - aMin);
```

Comment:
float everywhere here should be more than enough

Reply:
You're right. :+1:  I've changed it to float.